src/app/core/services/label.service.ts

```diff
@@ -22,17 +27,59 @@ const COLOR_WHITE = 'ffffff'; // Light color for text with dark background
  * from the GitHub repository for the WATcher application.
  */
 export class LabelService {
+  static readonly POLL_INTERVAL = 5000; // 5 seconds
+
   labels: Label[];
+  simplifiedLabels: simplifiedLabel[];
+
+  private labelsPollSubscription: Subscription;
+  private labelsSubject = new BehaviorSubject<simplifiedLabel[]>([]);
 
   constructor(private githubService: GithubService) {}
 
+  startPollLabels() {
+    if (this.labelsPollSubscription === undefined) {
```

Comment:
Might be clearer to users if we convert this to a guard. i.e.
```suggestion
    if (this.labelsPollSubscription) {
        return;
    }
```